### PR TITLE
USHIFT-1531: Loop and timeout for getting VM IP in scenarios

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

File: test/bin/scenario.sh [hunk 1]
```diff
@@ -147,12 +151,8 @@ launch_vm() {
...
 
     # Wait for an IP to be assigned
-    ip=$(get_vm_ip "${full_vmname}")
-    while [ -z "${ip}" ]; do
-        echo "Waiting for VM ${full_vmname} to have an IP"
-        sleep 30
-        ip=$(get_vm_ip "${full_vmname}")
-    done
+    echo "Waiting for VM ${full_vmname} to have an IP"
+    ip=$(timeout 300 bash -c "get_vm_ip ${full_vmname}")
```
Comment (dhellmann) [inline test/bin/scenario.sh:155]: We're using the same timeout everywhere else.

```suggestion
    ip=$(timeout "${VM_BOOT_TIMEOUT}" bash -c "get_vm_ip ${full_vmname}")
```

File: test/bin/scenario.sh [hunk 2]
```diff
@@ -88,10 +88,18 @@ prepare_kickstart() {
 # Show the IP address of the VM
 function get_vm_ip {
     local vmname="${1}"
-    sudo virsh domifaddr "${vmname}" \
-        | grep vnet \
-        | awk '{print $4}' \
-        | cut -f1 -d/
+    start=$(date +%s)
```
Comment (ggiguash) [inline test/bin/scenario.sh:91]: `vmname` and `start` can be `local -r`, right?

File: test/bin/scenario.sh [hunk 3]
```diff
@@ -88,10 +88,18 @@ prepare_kickstart() {
...
+    ip=$(bash -c "${ROOTDIR}/scripts/devenv-builder/manage-vm.sh ip -n ${vmname}")
```
Comment (ggiguash) [inline test/bin/scenario.sh:92]:
```suggestion
    local ip  
    ip=$("${ROOTDIR}/scripts/devenv-builder/manage-vm.sh" ip -n "${vmname}")
```

File: test/bin/scenario.sh (hunk 2 above)
Comment (pacevedom) [reply test/bin/scenario.sh:91]: Done.

File: test/bin/scenario.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (pacevedom) [reply test/bin/scenario.sh:92]: Done

Comment (pacevedom) [reply test/bin/scenario.sh:92]: Done.

File: test/bin/scenario.sh [hunk 4]
```diff
@@ -88,10 +88,18 @@ prepare_kickstart() {
...
+    while [ "${ip}" = "" ]; do
+        now=$(date +%s)
+        if [ $(( now - start )) -ge ${VM_BOOT_TIMEOUT} ]; then
+            echo "Timed out while waiting for IP retrieval"
+            exit 1
+        fi
+        sleep 1
+        ip=$(bash -c "${ROOTDIR}/scripts/devenv-builder/manage-vm.sh ip -n ${vmname}")
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply test/bin/scenario.sh:100]: Done.

**Pull request merged**